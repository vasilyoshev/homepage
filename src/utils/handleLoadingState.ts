import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
