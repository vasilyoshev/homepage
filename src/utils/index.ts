export * from './handleLoadingState';
export * from './testUtils';
