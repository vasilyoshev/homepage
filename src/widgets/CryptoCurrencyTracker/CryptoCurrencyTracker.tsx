import React, {ChangeEvent, useState, useEffect} from 'react';
import axios from 'axios';
import { Coin } from 'components';
import { CryptoCurrencyTrackerRes } from 'interfaces';
import styles from './CryptoCurrencyTracker.module.scss';

export const CryptoCurrencyTracker: React.FC = () => {
  const [coinsRes, setCoinsRes] = useState<Array<CryptoCurrencyTrackerRes>>([]);
  const [search, setSearch] = useState<string>('');

  useEffect(() => {
    axios.get('https://api.coingecko.com/api/v3/coins/markets?vs_currency=eur&order=market_cap_desc&per_page=5&page=1&sparkline=false')
      .then((res) => {
        setCoinsRes(res.data);
        console.log(res.data);
      })
      .catch((err) => console.log(err));
  },[]);

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setSearch(e.target.value);
  };

  const filteredCoins = coinsRes.filter((coin: CryptoCurrencyTrackerRes) => {
    return coin.name
      .toLowerCase()
      .includes(search.toLowerCase());
  });
  
  return (
    <div className={styles.coinApp}>
      <div className={styles.coinSearch}>
        <h1 className={styles.coinText}>Search a currency</h1>
        <form>
          <input 
            type='text'
            placeholder='Search' 
            className={styles.coinInput} 
            onChange={handleChange}
          />
        </form>
      </div> 
      {filteredCoins.map((coin) => {
        return ( 
          <Coin
            key={coin.id}
            name={coin.name} 
            image={coin.image}
            symbol={coin.symbol}
            marketcap={coin.market_cap}
            price={coin.current_price}
            priceChange={coin.price_change_percentage_24h}
            volume={coin.total_volume}
          />
        );
      })}
    </div>
  );
};
