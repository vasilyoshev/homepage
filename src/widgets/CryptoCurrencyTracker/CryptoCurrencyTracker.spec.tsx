import { render } from 'utils';
import { CryptoCurrencyTracker } from './CryptoCurrencyTracker';

describe('CryptoCurrencyTracker', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<CryptoCurrencyTracker />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
