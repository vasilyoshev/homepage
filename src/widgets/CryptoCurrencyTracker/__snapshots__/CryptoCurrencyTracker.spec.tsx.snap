// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCurrencyTracker renders successfully 1`] = `
<DocumentFragment>
  <div
    class="coinApp"
  >
    <div
      class="coinSearch"
    >
      <h1
        class="coinText"
      >
        Search a currency
      </h1>
      <form>
        <input
          class="coinInput"
          placeholder="Search"
          type="text"
        />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
